request_router.cc

```diff
@@ -0,0 +1,103 @@
+#include "request_router.h"
+#include "request_handlers.h"
+
+#include <boost/tokenizer.hpp>
+
+// TODO: Split out HTTP parsing into its own class
+bool RequestRouter::routeRequest(const ServerConfig& server_config,
+                                 const char* request_buffer,
+                                 const size_t& request_buffer_size,
+                                 char* &response_buffer,
+                                 size_t& response_buffer_size) {
+
```

Comment:
Reviews for this file might be covered/the same as in the server-static-files branch